lib/streamlit/runtime/state/query_params.py

```diff
@@ -91,8 +91,9 @@ def clear(self) -> None:
         self._query_params.clear()
         self._send_query_param_msg()
 
-    def to_dict(self) -> Dict[str, Union[List[str], str]]:
-        return self._query_params
+    def to_dict(self) -> Dict[str, str]:
```

Comment:
Could we mimick [the `to_dict` method as Flask provides it?](https://werkzeug.palletsprojects.com/en/3.0.x/datastructures/#werkzeug.datastructures.MultiDict.to_dict)

For `flat=True` it would do what it currently does. For `flat=False` it would make a list for every key, regardless of whether it's only one value. I like this because if the `flat` parameter is not available the user of `to_dict` will have to do a lot of `isinstance` checking.

Or is there already a different way of predictably getting a list of values (even if there is only one value)?

Reply:
Should be solved with `get_all()` support.

Reply:
> Should be solved with `get_all()` support.

Thanks! I'll check it out